ota-plus-web/app/reactapp-new/src/store/software/reducers.ts

```diff
@@ -0,0 +1,21 @@
+import {
+  ISoftwareState,
+  ActionType,
+} from './types';
+import { Actions } from './actions';
+
+export const initialState: ISoftwareState = {
+  versionsTotal: 0
+};
+
+export const softwareReducer = (state = initialState, action: Actions): ISoftwareState => {
+  switch (action.type) {
+    case ActionType.SET_ALL_SOFTWARE_DONE: {
+      return {
+        ...state, ...action.payload
```

Comment:
I think we can just return action.payload here

Reply:
Done